apps/pollbook/backend/src/local_store.ts

```diff
@@ -676,23 +693,124 @@ export class LocalStore extends Store {
+   *   When a specific party is provided, only voters registered with that party are included
+   *   in the statistics for total voters and new registrations. Only check-ins with a ballot party matching the
+   *   provided filter are counted for total check ins and total absentee check ins. That includes check ins from a
+   *   voter matching the party filter and undeclared voters who checked in with a ballot party matching the filter.
+   *   When the partyFilter is 'UND' the number of undeclared voters who checked in with a democratic or republican ballot
+   *   party is returned in totalUndeclaredDemCheckIns and totalUndeclaredRepCheckIns respectively.
+   * @returns Summary statistics for the filtered set of voters.
+   */
+  getPrimarySummaryStatistics(
+    partyFilter: PartyFilterAbbreviation
+  ): PrimarySummaryStatistics {
+    const { configuredPrecinctId } = this.getPollbookConfigurationInformation();
+    const election = this.getElection();
+    assert(election !== undefined);
+    assert(election.type === 'primary');
+    const voters = this.getAllVoters({
+      matchConfiguredPrecinctId: !!configuredPrecinctId,
+    });
+    assert(voters);
+    const votersMatchingParty =
+      partyFilter === 'ALL'
+        ? voters
+        : Object.fromEntries(
+            Object.entries(voters).filter(
+              ([, voter]) => voter.party === partyFilter
+            )
+          );
+    const undeclaredDemVoters =
+      partyFilter === 'UND'
+        ? Object.values(votersMatchingParty).filter(
+            (voter) => voter.checkIn && voter.checkIn.ballotParty === 'DEM'
+          ).length
+        : 0;
+    const undeclaredRepVoters =
+      partyFilter === 'UND'
+        ? Object.values(votersMatchingParty).filter(
+            (voter) => voter.checkIn && voter.checkIn.ballotParty === 'REP'
+          ).length
+        : 0;
+    // We have to consider ALL voters here as undeclared party voters may check in with a party
+    const votersWithCheckInMatchingParty = Object.fromEntries(
+      Object.entries(voters).filter(
+        ([, voter]) =>
+          voter.checkIn &&
+          (partyFilter === 'ALL' || voter.checkIn.ballotParty === partyFilter)
+      )
+    );
+    const totalVoters = Object.keys(votersMatchingParty).length;
+    const totalAbsenteeCheckIns = Object.values(
+      votersWithCheckInMatchingParty
+    ).filter((v) => v.checkIn && v.checkIn.isAbsentee).length;
+    const totalNewRegistrations = Object.values(votersMatchingParty).filter(
       (v) => v.registrationEvent !== undefined
     ).length;
-    const totalCheckIns = Object.values(voters).filter((v) => v.checkIn).length;
+    const totalCheckIns = Object.values(votersWithCheckInMatchingParty).length;
 
     return {
       totalVoters,
       totalCheckIns,
       totalAbsenteeCheckIns,
       totalNewRegistrations,
+      totalUndeclaredDemCheckIns: undeclaredDemVoters,
+      totalUndeclaredRepCheckIns: undeclaredRepVoters,
```

Comment:
It feels a little counterintuitive that when the filter == `ALL` this property reports 0 undeclared REP/DEM check-ins. The behavior for `UND` might be different enough that it makes sense to split this into 2 methods eg `getPrimarySummaryStatistics` where `partyFilter: Omit<PartyFilterAbbreviation, 'UND'>` and `getPrimarySummaryStatisticsForUndeclaredVoters` and have the API coordinate the appropriate call. Thoughts?